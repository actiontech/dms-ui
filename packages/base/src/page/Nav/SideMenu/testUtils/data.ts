export const mockRecentlyOpenedProjectsData = {
  recentlyProjects: [
    {
      project_id: '1',
      project_name: 'test_project'
    }
  ],
  updateRecentlyProject: jest.fn(),
  currentProjectID: '0'
};
