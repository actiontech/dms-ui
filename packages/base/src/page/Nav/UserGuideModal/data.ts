import { GetUserSystemEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import { t } from '../../../locale';

export const systemDictionary = {
  [GetUserSystemEnum.WORKBENCH]: t('dmsMenu.userGuide.sqlWorkbench.label'),
  [GetUserSystemEnum.MANAGEMENT]: t('dmsMenu.userGuide.adminPanel.label')
};
