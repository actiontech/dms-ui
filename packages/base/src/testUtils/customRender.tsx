import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/system';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { storeFactory } from '@actiontech/shared/lib/testUtil/mockRedux';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

export type RenderParams = Parameters<typeof render>;

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={themeData}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
