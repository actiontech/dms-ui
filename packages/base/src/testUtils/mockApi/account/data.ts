export const mockAccessTokenInfo = {
  access_token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHA',
  is_expired: false,
  token_expired_timestamp: '2024-03-31T10:25:10+08:00'
};
