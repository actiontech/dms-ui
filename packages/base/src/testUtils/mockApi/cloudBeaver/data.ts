export const disableSqlQueryUrlData = {
  enable_sql_query: false,
  sql_query_root_uri: '/sql_query'
};

export const enableSqlQueryUrlData = {
  enable_sql_query: true,
  sql_query_root_uri: '/new_sql_query'
};
