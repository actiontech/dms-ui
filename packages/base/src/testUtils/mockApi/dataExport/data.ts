import {
  IAddDataExportTaskReply,
  IAddDataExportWorkflowReply,
  IListDataExportWorkflow
} from '@actiontech/shared/lib/api/base/service/common';
import { ListDataExportWorkflowStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export const DataExportWorkflowList: IListDataExportWorkflow[] = [
  {
    project_uid: '700300',
    workflow_uid: '1752172808714063872',
    workflow_name: 'mysql-1_20240130113114',
    desc: 'desc',
    creater: {
      uid: '700200',
      name: 'admin'
    },
    created_at: '2024-01-30T11:32:06.053+08:00',
    exported_at: '2024-01-30T11:33:06.053+08:00',
    status: ListDataExportWorkflowStatusEnum.finish,
    current_step_assignee_user_list: [],
    current_step_type: ''
  },
  {
    project_uid: '700300',
    workflow_uid: '1752172808714063871',
    workflow_name: 'mysql-2_20240130113114',
    desc: 'desc',
    creater: {
      uid: '700200',
      name: 'admin'
    },
    created_at: '2024-01-30T11:32:06.053+08:00',
    exported_at: '2024-01-30T11:33:06.053+08:00',
    status: ListDataExportWorkflowStatusEnum.wait_for_approve,
    current_step_assignee_user_list: [
      {
        uid: '700200',
        name: 'admin'
      }
    ],
    current_step_type: ''
  }
];

export const AddDataExportTaskResponseData: IAddDataExportTaskReply['data'] = {
  data_export_task_uids: ['1752522203951271936']
};

export const AddDataExportWorkflowResponseData: IAddDataExportWorkflowReply['data'] =
  {
    export_data_workflow_uid: '1483489234842342'
  };
