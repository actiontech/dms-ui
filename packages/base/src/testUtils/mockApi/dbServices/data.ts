import {
  ICheckDBServiceIsConnectableReplyItem,
  IListDBService,
  IListDBServiceTipItem,
  IListGlobalDBService
} from '@actiontech/shared/lib/api/base/service/common';
import { SQLQueryConfigAllowQueryWhenLessThanAuditLevelEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export const dbServices: IListDBService[] = [
  {
    business: 'business',
    db_type: 'MySQL',
    host: '127.0.0.1',
    maintenance_times: [],
    name: 'test',
    password: 'root',
    port: '3306',
    uid: '123123',
    user: 'root',
    sqle_config: {
      rule_template_id: '1',
      rule_template_name: 'rule_template_name1',
      sql_query_config: {
        allow_query_when_less_than_audit_level:
          SQLQueryConfigAllowQueryWhenLessThanAuditLevelEnum.notice,
        audit_enabled: true
      }
    },
    instance_audit_plan_id: 1232,
    audit_plan_types: [
      { type: 'mysql_slow_log', desc: '慢日志', audit_plan_id: 1 }
    ]
  },
  {
    business: 'business',
    db_type: 'MySQL',
    host: 'localhost',
    maintenance_times: [],
    name: 'test2',
    password: '123',
    port: '3306',
    uid: '300123',
    user: 'test1'
  }
];

export const dbServicesTips: IListDBServiceTipItem[] = [
  {
    db_type: 'MySQL',
    host: '127.0.0.1',
    name: 'test',
    port: '3306',
    id: '123123'
  },
  {
    db_type: 'MySQL',
    host: 'localhost',
    name: 'test2',
    port: '3306',
    id: '300123'
  }
];

export const checkConnectableReply: ICheckDBServiceIsConnectableReplyItem[] = [
  {
    component: 'sqle',
    is_connectable: true
  },
  {
    component: 'provision',
    is_connectable: false
  }
];

export const globalDataSourceMockData: IListGlobalDBService[] = [
  {
    uid: '123456',
    name: 'vm1-mysql',
    db_type: 'MySQL',
    host: '10.186.62.41',
    port: '3306',
    business: 'vm1',
    maintenance_times: [
      {
        maintenance_start_time: {
          hour: 2,
          minute: 0
        },
        maintenance_stop_time: {
          hour: 8,
          minute: 0
        }
      }
    ],
    desc: '222',
    source: 'SQLE',
    project_uid: '700300',
    is_enable_masking: false,
    project_name: 'default',
    unfinished_workflow_num: 1,
    is_enable_audit: true
  },
  {
    uid: '123457',
    name: 'vm1-mysql',
    db_type: 'MySQL',
    host: '',
    port: '3306',
    business: 'test',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '1811294305910788096',
    is_enable_masking: false,
    project_name: 'project1',
    unfinished_workflow_num: 0,
    is_enable_audit: false
  },
  {
    uid: '123458',
    name: 'vm1-oracle',
    db_type: 'Oracle',
    host: '10.186.62.41',
    port: '',
    business: 'vm1',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '700300',
    is_enable_masking: false,
    project_name: 'default',
    unfinished_workflow_num: 0,
    is_enable_audit: false
  }
];

export const globalDBServicesTipsMockData = {
  db_type: ['MySQL']
};
