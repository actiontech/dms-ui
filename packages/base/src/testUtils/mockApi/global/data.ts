import {
  IGetUser,
  IGetOauth2TipsResData,
  IBasicInfo,
  ICompanyNotice,
  IListDBService
} from '@actiontech/shared/lib/api/base/service/common';
import {
  GetUserAuthenticationTypeEnum,
  ListDBServiceLastConnectionTestStatusEnum
} from '@actiontech/shared/lib/api/base/service/common.enum';
import { SupportLanguage } from '@actiontech/shared/lib/enum';

export const UserInfo = {
  token: 'Dhsiwkow133jn',
  name: 'test',
  userUid: '300123'
};

export const GetUserPayload: IGetUser = {
  authentication_type: GetUserAuthenticationTypeEnum.dms,
  email: 'test@gamil.com',
  is_admin: true,
  name: 'test',
  language: SupportLanguage.zhCN,
  uid: '300123',
  user_bind_projects: [
    {
      is_manager: true,
      project_id: '700300',
      project_name: 'default'
    }
  ]
};

export const oauth2Tips: IGetOauth2TipsResData = {
  enable_oauth2: true,
  login_tip: 'Login With Oauth2'
};

export const BasicInfoMockData: IBasicInfo = {
  components: [],
  logo_url: '/v1/dms/personalization/logo',
  title: 'Actiontech'
};

export const CompanyNoticeMockData: ICompanyNotice = {
  notice_str: 'notice',
  read_by_current_user: false
};

export const DBServicesList: IListDBService[] = [
  {
    uid: '1739531854064652288',
    name: 'mysql-1',
    db_type: 'MySQL',
    host: '10.186.62.13',
    port: '33061',
    user: 'root',
    password: 'Zgl4cTg5xeIq9c/pkc8Y5A==',
    business: 'test',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '700300',
    sqle_config: {
      rule_template_name: 'default_MySQL',
      rule_template_id: '1',
      sql_query_config: {
        max_pre_query_rows: 0,
        query_timeout_second: 0,
        audit_enabled: false
      }
    },
    is_enable_masking: true,
    instance_audit_plan_id: 1232,
    audit_plan_types: [
      { type: 'mysql_slow_log', desc: '慢日志', audit_plan_id: 1 }
    ],
    last_connection_test_error_message: '',
    last_connection_test_status:
      ListDBServiceLastConnectionTestStatusEnum.connect_success,
    last_connection_test_time: '2024-11-15T15:05:10.175+08:00',
    enable_backup: true
  },
  {
    uid: '1739531942258282496',
    name: 'mysql-2',
    db_type: 'MySQL',
    host: '10.186.62.13',
    port: '33062',
    user: 'root',
    password: 'Zgl4cTg5xeIq9c/pkc8Y5A==',
    business: 'test-2',
    maintenance_times: [],
    desc: '',
    source: 'SQLE',
    project_uid: '700300',
    sqle_config: {
      rule_template_name: 'default_MySQL',
      rule_template_id: '1',
      sql_query_config: {
        max_pre_query_rows: 0,
        query_timeout_second: 0,
        audit_enabled: false
      }
    },
    is_enable_masking: false,
    last_connection_test_error_message: 'error message',
    last_connection_test_status:
      ListDBServiceLastConnectionTestStatusEnum.connect_failed,
    last_connection_test_time: '2024-11-15T15:05:10.175+08:00',
    enable_backup: false
  }
];

export const mockVerifyLoginData = {
  verify_failed_msg: '',
  user_uid: '700200',
  two_factor_enabled: false
};

export const maskRuleData = [
  {
    effect: '182****8767',
    id: 1,
    description: '手机号码',
    masking_type: '电话',
    reference_fields: ['电话', '投诉电话']
  },
  {
    id: 2,
    masking_type: '护照',
    reference_fields: ['passport', '护照']
  },
  {
    effect: '182****8780',
    id: 3,
    reference_fields: ['SSN']
  },
  {
    effect: '182****8790',
    id: 4
  }
];
