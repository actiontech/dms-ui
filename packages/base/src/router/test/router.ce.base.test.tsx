/**
 * @test_version ce
 */

import { MemoryRouterProps } from 'react-router-dom';
import { act, screen, cleanup } from '@testing-library/react';

import mockUseRoutes, { RenderRouterComponent } from './data';
import { superRender } from '../../testUtils/customRender';
import { mockProjectInfo } from '@actiontech/shared/lib/testUtil/mockHook/data';

describe('base/router-base-ce', () => {
  const projectID = mockProjectInfo.projectID;

  const customRender = (
    initialEntries: MemoryRouterProps['initialEntries'] = []
  ) => {
    return superRender(<RenderRouterComponent type="base" />, undefined, {
      routerProps: {
        initialEntries
      }
    });
  };

  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
    cleanup();
  });

  it('render base route data snap', () => {
    expect(mockUseRoutes('base')).toMatchSnapshot();
  });

  describe('render base router when version is ce', () => {
    it('render router home', async () => {
      const { baseElement } = customRender(['/']);

      expect(screen.getByText('home')).toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });

    it('render router userCenter', async () => {
      const { baseElement } = customRender(['/userCenter']);

      expect(screen.getByText('userCenter')).toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });

    it('render router system', async () => {
      const { baseElement } = customRender(['/system']);

      expect(screen.getByText('system')).toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });

    it('render router account', async () => {
      const { baseElement } = customRender(['/account']);

      expect(screen.getByText('account')).toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });

    describe('render router project', () => {
      it('render router cloudBeaver', async () => {
        const { baseElement } = customRender([
          `/project/${projectID}/cloudBeaver`
        ]);

        await act(async () => jest.advanceTimersByTime(0));
        expect(screen.getByText('cloudBeaver')).toBeInTheDocument();
        expect(baseElement).toMatchSnapshot();
      });

      it('render route member', async () => {
        const { baseElement } = customRender([`/project/${projectID}/member`]);

        await act(async () => jest.advanceTimersByTime(0));
        expect(screen.getByText('member')).toBeInTheDocument();
        expect(baseElement).toMatchSnapshot();
      });

      it('render route dataSourceList', async () => {
        const { baseElement } = customRender([
          `/project/${projectID}/db-services`
        ]);

        await act(async () => jest.advanceTimersByTime(0));
        expect(baseElement).toMatchSnapshot();
      });

      it('render route dataSourceCreate', async () => {
        const { baseElement } = customRender([
          `/project/${projectID}/db-services/create`
        ]);

        await act(async () => jest.advanceTimersByTime(0));
        expect(screen.getByText('dataSourceCreate')).toBeInTheDocument();
        expect(baseElement).toMatchSnapshot();
      });

      it('render route dataSourceUpdate', async () => {
        const { baseElement } = customRender([
          `/project/${projectID}/db-services/update/dbServiceUid`
        ]);

        await act(async () => jest.advanceTimersByTime(0));
        expect(screen.getByText('dataSourceUpdate')).toBeInTheDocument();
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
