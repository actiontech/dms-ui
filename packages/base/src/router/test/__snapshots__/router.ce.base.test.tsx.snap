// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base/router-base-ce render base route data snap 1`] = `
[
  {
    "element": <div>
      home
    </div>,
    "key": "home",
    "path": "/",
  },
  {
    "element": <div>
      userCenter
    </div>,
    "key": "userCenter",
    "path": "/user-center",
  },
  {
    "element": <div>
      system
    </div>,
    "key": "system",
    "path": "/system",
  },
  {
    "element": <div>
      account
    </div>,
    "key": "account",
    "path": "/account",
  },
  {
    "element": <div>
      dataSourceManagement
    </div>,
    "key": "dataSourceManagement",
    "path": "/data-source-management",
  },
  {
    "element": <div>
      transit
    </div>,
    "key": "transit",
    "path": "/transit",
  },
  {
    "children": [
      {
        "element": <div>
          member
        </div>,
        "key": "member",
        "path": ":projectID/member",
      },
      {
        "children": [
          {
            "element": <div>
              dataSourceList
            </div>,
            "key": "dataSourceList",
            "path": "",
          },
          {
            "element": <div>
              dataSourceCreate
            </div>,
            "key": "dataSourceCreate",
            "path": "create",
          },
          {
            "element": <div>
              dataSourceUpdate
            </div>,
            "key": "dataSourceUpdate",
            "path": "update/:dbServiceUid",
          },
        ],
        "element": <div
          data-test-parent-id="dataSource"
        >
          <Outlet />
        </div>,
        "key": "dataSource",
        "path": ":projectID/db-services",
      },
      {
        "element": <div>
          dataExportManagement
        </div>,
        "key": "dataExportManagement",
        "path": ":projectID/data/export",
      },
      {
        "element": <div>
          cloudBeaver
        </div>,
        "key": "cloudBeaver",
        "path": ":projectID/cloud-beaver",
      },
    ],
    "element": <div
      data-test-parent-id="projectDetail"
    >
      <Outlet />
    </div>,
    "key": "projectDetail",
    "path": "/project/*",
  },
  {
    "element": <div>
      reportStatistics
    </div>,
    "key": "reportStatistics",
    "path": "/sqle/report-statistics",
  },
  {
    "element": <div>
      rule
    </div>,
    "key": "rule",
    "path": "/sqle/rule",
  },
  {
    "element": <div>
      globalRuleTemplateDetail
    </div>,
    "key": "globalRuleTemplateDetail",
    "path": "/sqle/rule-manager/global-detail/:templateName/:dbType",
  },
  {
    "children": [
      {
        "element": <div>
          ruleTemplateManagement
        </div>,
        "key": "ruleTemplateManagement",
        "path": "",
      },
      {
        "element": <div>
          globalRuleTemplateCreate
        </div>,
        "key": "globalRuleTemplateCreate",
        "path": "global-create",
      },
      {
        "element": <div>
          globalRuleTemplateImport
        </div>,
        "key": "globalRuleTemplateImport",
        "path": "global-import",
      },
      {
        "element": <div>
          globalRuleTemplateUpdate
        </div>,
        "key": "globalRuleTemplateUpdate",
        "path": "global-update/:templateName",
      },
      {
        "element": <div>
          createCustomRule
        </div>,
        "key": "createCustomRule",
        "path": "custom-create",
      },
      {
        "element": <div>
          updateCustomRule
        </div>,
        "key": "updateCustomRule",
        "path": "custom-update/:ruleID",
      },
    ],
    "element": <div
      data-test-parent-id="ruleManager"
    >
      <Outlet />
    </div>,
    "key": "ruleManager",
    "path": "/sqle/rule-manager",
  },
  {
    "children": [
      {
        "element": <div>
          knowledgeIndex
        </div>,
        "key": "knowledgeIndex",
        "path": "",
      },
    ],
    "element": <div
      data-test-parent-id="knowledge"
    >
      <Outlet />
    </div>,
    "key": "knowledge",
    "path": "/sqle/knowledge",
  },
  {
    "element": <div>
      globalDashboard
    </div>,
    "key": "globalDashboard",
    "path": "/sqle/global-dashboard",
  },
  {
    "children": [
      {
        "element": <div>
          projectOverview
        </div>,
        "key": "projectOverview",
        "path": ":projectID/overview",
      },
      {
        "children": [
          {
            "element": <div>
              sqlExecWorkflowList
            </div>,
            "key": "sqlExecWorkflowList",
            "path": "",
          },
          {
            "element": <div>
              createSqlExecWorkflow
            </div>,
            "key": "createSqlExecWorkflow",
            "path": "create",
          },
          {
            "element": <div>
              workflowAnalyze
            </div>,
            "key": "workflowAnalyze",
            "path": ":taskId/:sqlNum/analyze",
          },
          {
            "element": <div>
              workflowDetail
            </div>,
            "key": "workflowDetail",
            "path": ":workflowId",
          },
          {
            "element": <div>
              workflowSqlFileStatementOverview
            </div>,
            "key": "workflowSqlFileStatementOverview",
            "path": ":taskId/files/:fileId/sqls",
          },
        ],
        "element": <div
          data-test-parent-id="sqlExecWorkflow"
        >
          <Outlet />
        </div>,
        "key": "sqlExecWorkflow",
        "path": ":projectID/exec-workflow",
      },
      {
        "children": [
          {
            "element": <div>
              sqlAuditList
            </div>,
            "key": "sqlAuditList",
            "path": "",
          },
          {
            "element": <div>
              sqlAuditCreate
            </div>,
            "key": "sqlAuditCreate",
            "path": "create",
          },
          {
            "element": <div>
              sqlAuditDetail
            </div>,
            "key": "sqlAuditDetail",
            "path": "detail/:sql_audit_record_id",
          },
          {
            "element": <div>
              sqlAuditCreateOptimization
            </div>,
            "key": "sqlAuditCreateOptimization",
            "path": "create-optimization",
          },
          {
            "element": <div>
              sqlOptimizationResult
            </div>,
            "key": "sqlOptimizationResult",
            "path": "optimization-result/:optimizationId",
          },
        ],
        "element": <div
          data-test-parent-id="sqlAudit"
        >
          <Outlet />
        </div>,
        "key": "sqlAudit",
        "path": ":projectID/sql-audit",
      },
      {
        "children": [
          {
            "element": <div>
              SqlManagementConfList
            </div>,
            "key": "SqlManagementConfList",
            "path": "",
          },
          {
            "element": <div>
              CreateSqlManagementConf
            </div>,
            "key": "CreateSqlManagementConf",
            "path": "create",
          },
          {
            "element": <div>
              UpdateSqlManagementConf
            </div>,
            "key": "UpdateSqlManagementConf",
            "path": "update/:id",
          },
          {
            "element": <div>
              SqlManagementConfDetail
            </div>,
            "key": "SqlManagementConfDetail",
            "path": ":id",
          },
          {
            "element": <div>
              SqlManagementConfAnalyze
            </div>,
            "key": "SqlManagementConfAnalyze",
            "path": ":instanceAuditPlanId/analyze/:id",
          },
        ],
        "element": <div
          data-test-parent-id="sqlManagementConf"
        >
          <Outlet />
        </div>,
        "key": "sqlManagementConf",
        "path": ":projectID/sql-management-conf",
      },
      {
        "element": <div>
          sqlInsights
        </div>,
        "key": "sqlInsights",
        "path": ":projectID/sql-insights",
      },
      {
        "children": [
          {
            "element": <div>
              SqlManagement
            </div>,
            "key": "SqlManagement",
            "path": "",
          },
        ],
        "element": <div
          data-test-parent-id="sqlManagement"
        >
          <Outlet />
        </div>,
        "key": "sqlManagement",
        "path": ":projectID/sql-management",
      },
      {
        "element": <div>
          dashboard
        </div>,
        "key": "dashboard",
        "path": ":projectID/dashboard",
      },
      {
        "children": [
          {
            "element": <div>
              ruleTemplate
            </div>,
            "key": "ruleTemplate",
            "path": "",
          },
          {
            "element": <div>
              ruleTemplate
            </div>,
            "key": "ruleTemplate",
            "path": "list",
          },
          {
            "element": <div>
              ruleTemplateCreate
            </div>,
            "key": "ruleTemplateCreate",
            "path": "create",
          },
          {
            "element": <div>
              ruleTemplateImport
            </div>,
            "key": "ruleTemplateImport",
            "path": "import",
          },
          {
            "element": <div>
              ruleTemplateDetail
            </div>,
            "key": "ruleTemplateDetail",
            "path": "detail/:templateName/:dbType",
          },
          {
            "element": <div>
              ruleTemplateUpdate
            </div>,
            "key": "ruleTemplateUpdate",
            "path": "update/:templateName",
          },
        ],
        "element": <div
          data-test-parent-id="ruleTemplate"
        >
          <Outlet />
        </div>,
        "key": "ruleTemplate",
        "path": ":projectID/rule/template",
      },
      {
        "children": [
          {
            "element": <div>
              progressDetail
            </div>,
            "key": "progressDetail",
            "path": "",
          },
        ],
        "element": <div
          data-test-parent-id="progress"
        >
          <Outlet />
        </div>,
        "key": "progress",
        "path": ":projectID/progress",
      },
      {
        "element": <div>
          Whitelist
        </div>,
        "key": "Whitelist",
        "path": ":projectID/whitelist",
      },
      {
        "element": <div>
          sqlManagementException
        </div>,
        "key": "sqlManagementException",
        "path": ":projectID/sql-management-exception",
      },
      {
        "element": <div>
          operationRecord
        </div>,
        "key": "operationRecord",
        "path": ":projectID/operation-record",
      },
      {
        "children": [
          {
            "element": <div>
              pluginAuditList
            </div>,
            "key": "pluginAuditList",
            "path": "",
          },
        ],
        "element": <div
          data-test-parent-id="pluginAudit"
        >
          <Outlet />
        </div>,
        "key": "pluginAudit",
        "path": ":projectID/plugin-audit",
      },
      {
        "element": <div>
          pushRuleConfiguration
        </div>,
        "key": "pushRuleConfiguration",
        "path": ":projectID/push-rule",
      },
      {
        "children": [
          {
            "element": <div>
              pipelineConfigurationList
            </div>,
            "key": "pipelineConfigurationList",
            "path": "",
          },
          {
            "element": <div>
              createPipelineConfiguration
            </div>,
            "key": "createPipelineConfiguration",
            "path": "create",
          },
          {
            "element": <div>
              updatePipelineConfiguration
            </div>,
            "key": "updatePipelineConfiguration",
            "path": "update/:id",
          },
        ],
        "element": <div
          data-test-parent-id="pipelineConfiguration"
        >
          <Outlet />
        </div>,
        "key": "pipelineConfiguration",
        "path": ":projectID/pipeline-configuration",
      },
      {
        "children": [
          {
            "element": <div>
              VersionManagementList
            </div>,
            "key": "VersionManagementList",
            "path": "",
          },
        ],
        "element": <div
          data-test-parent-id="versionManagement"
        >
          <Outlet />
        </div>,
        "key": "versionManagement",
        "path": ":projectID/version-management",
      },
      {
        "element": <div>
          dataSourceComparison
        </div>,
        "key": "dataSourceComparison",
        "path": ":projectID/data-source-comparison",
      },
      {
        "element": <div>
          projectRedirect
        </div>,
        "key": "projectRedirect",
        "path": "*",
      },
    ],
    "element": <div
      data-test-parent-id="sqleProjectDetail"
    >
      <Outlet />
    </div>,
    "key": "sqleProjectDetail",
    "path": "/sqle/project/*",
  },
  {
    "element": <div>
      notFound
    </div>,
    "key": "notFound",
    "path": "*",
  },
]
`;

exports[`base/router-base-ce render base router when version is ce render router account 1`] = `
<body>
  <div>
    <div>
      account
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router home 1`] = `
<body>
  <div>
    <div>
      home
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router project render route dataSourceCreate 1`] = `
<body>
  <div>
    <div
      data-test-parent-id="projectDetail"
    >
      <div
        data-test-parent-id="dataSource"
      >
        <div>
          dataSourceCreate
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router project render route dataSourceList 1`] = `
<body>
  <div>
    <div
      data-test-parent-id="projectDetail"
    >
      <div
        data-test-parent-id="dataSource"
      >
        <div>
          dataSourceList
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router project render route dataSourceUpdate 1`] = `
<body>
  <div>
    <div
      data-test-parent-id="projectDetail"
    >
      <div
        data-test-parent-id="dataSource"
      >
        <div>
          dataSourceUpdate
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router project render route member 1`] = `
<body>
  <div>
    <div
      data-test-parent-id="projectDetail"
    >
      <div>
        member
      </div>
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router project render router cloud-beaver 1`] = `
<body>
  <div>
    <div
      data-test-parent-id="projectDetail"
    >
      <div>
        cloudBeaver
      </div>
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router system 1`] = `
<body>
  <div>
    <div>
      system
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router transit 1`] = `
<body>
  <div>
    <div>
      transit
    </div>
  </div>
</body>
`;

exports[`base/router-base-ce render base router when version is ce render router user-center 1`] = `
<body>
  <div>
    <div>
      userCenter
    </div>
  </div>
</body>
`;
